### PR TITLE
Changes to --users and --groups

### PR DESCRIPTION
**Pull request by aj-cgtech**
users() was failing on a bad attribute, changed code to use getattr
instead. If attribute is missing, it no longer throws exception.

extraction of domain from distinguished name was not working in all
circumstances. FOO.COM would work, but FOO.CO.UK or even FOO.BAR.CO.UK
would extract CO incorrectly. function now extracts fully qualified
domain, which then gets shortened by db_add_user() function.

### CODE REVIEW

**Comment by aj-cgtech** [comment]
I've added the changes for issue #236 as suggested.

The only thing I would say, is that in this change I have loaded the ConfigParser individually in each protocol class. I would have preferred to only change it once in context.py and passed the object, but I couldn't work out how the classes were instantiated, or how to pass the already instantiated context object to them.

**Comment by byt3bl33d3r** [comment]
@aj-cgtech Merged with some changes. Thanks again!

So the Context object is not passed to each protocol object rather it's just a placeholder object and it's passed to [each module](https://github.com/byt3bl33d3r/CrackMapExec/blob/master/cme/connection.py#L94-L108) (hence the context parameter)

The ConfigParser object is actually passed individually to each object using `setattr` [here](https://github.com/byt3bl33d3r/CrackMapExec/blob/master/cme/crackmapexec.py#L137)

The Protocol classes are instantiated [here](https://github.com/byt3bl33d3r/CrackMapExec/blob/master/cme/crackmapexec.py#L123-L135), the logic behind how that works is in [protocol_loader.py](https://github.com/byt3bl33d3r/CrackMapExec/blob/master/cme/loaders/protocol_loader.py)

Basically all I did was add a entry in the config file to customize the pwn3d label :+1:
